packages/@tinacms/cli/src/cmds/audit/validations/validateRichText.ts

```diff
@@ -0,0 +1,45 @@
+/**
+Copyright 2021 Forestry.io Holdings, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { AuditIssue, AuditWarning } from '../issue'
+
+const validateValues = (values, addWarning: (message: string) => void) => {
+  let fields = Object.keys(values)
+    .map((fieldName) => values[fieldName])
+    .filter((field) => typeof field === 'object')
+
+  fields.forEach((field) => {
```

Comment:
This isn't the best way to iterate through recursively because for `object` fields with `templates` this won't work. Whenever we need to iterate, we always use the schema to do it. I'm not sure if this is the best approach or if it'd just be easier to throw a special error during `audit` rather than catching things in the `invalid_markdown` node.

Reply:
> This isn't the best way to iterate through recursively because for object fields with templates this won't work

This is just searching through a document's `_values` (not actually searching on a "fields" object). Wouldn't a response setup with templates vs fields still output the same response object?

I've kind of just been going off of what the backend API response actually gives us, but if it could provide a root-level list of errors that would be easier to handle in the CLI.

Reply:
Yeah if it works that's fine. It's definitely feeling a bit of a weird API to step through rather than just through a special kind of error and catching it higher up. Is this also handling `rich-text` inside an array of values?

Reply:
Hmm, okay well I see a few paths forward (let me know if you can think of others).
1) We can remove the rich-text check from the audit for now. 
2) We can take the time now to revamp how errors are handled on the backend, so a document-list request can return a root-level errors list, while still providing a valid response. That way we can cleanly handle these in the CLI, and the document list. I'm handle to poke around and see if I can figure this out, but you may be able to navigate the GraphQL code quicker if you have bandwidth.

Reply:
I definitely think the whole query blowing up on an error in one of the documents should be fixed up. I can tackle that this week. And if we did that, perhaps we can just allow the `invalid_markdown` error to bubble up only during `audit`. That would allow audit to work and just show the error on a document level I think.

Reply:
Would it be possible to return partial data with an errors array for all requests? (It seems that when an error is thrown from the resolver, the full response comes back null) 

I could also see this type of behaviour being useful to fetch/show our document list view, but show "alert" icons next to the documents with issues.

Reply:
@jeffsee55 I logged an internal ticket to capture how this PR is blocked: https://linear.app/tina/issue/ENG-506/surface-errors-in-audit-that-arent-breaking-in-the-cms